### PR TITLE
[Filebeat] Get AWS credentials again when token expired

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR is to add `GetAWSCredentials` function to get credentials again when previous AWS credentials are expired or invalid.

## Why is it important?

This will avoid users have to manually stop and restart Filebeat in order to obtain the new AWS credentials.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~

## How to test this PR locally

Use `~/.aws/credentials` to store AWS credentials and then in Filebeat, change filebeat.yml to use s3 input:
```
filebeat.inputs:
- type: s3
  queue_url:   https://sqs.us-east-1.amazonaws.com/428152502467/test-fb-ks
  credential_profile_name: elastic-beats
```

Before starting Filebeat, when generating temporary credentials, set `DurationSeconds` to be 900 seconds. Wait after 900 seconds(15min) and then start filebeat with `./filebeat -e`. You should see error message immediately:
```
2020-03-24T11:51:37.872-0600    WARN    [s3]    s3/input.go:206 credentials are invalid, acquiring AWS credentials again from config: ExpiredToken: The security token included in the request is expired
        status code: 403, request id: 702080a4-8d34-5d5a-b212-d650c7ddfa53
```
Then generate a new set of temporary credentials and update credentials in the file `~/.aws/credentials`. This error message should disappear right away.

## Related issues

- Closes https://github.com/elastic/beats/issues/17189

### CODE REVIEW

File: x-pack/filebeat/input/s3/input.go [hunk 1]
```diff
@@ -178,30 +178,45 @@ func (p *s3Input) Run() {
...
 		output, err := p.receiveMessage(svcSQS, visibilityTimeout)
 		if err != nil {
-			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == awssdk.ErrCodeRequestCanceled {
-				continue
+			if awsErr, ok := err.(awserr.Error); ok {
+				if awsErr.Code() == awssdk.ErrCodeRequestCanceled {
+					continue
+				}
+
+				if awsErr.Code() == "InvalidClientTokenId" || awsErr.Code() == "ExpiredToken" {
```
Comment (exekias) [inline x-pack/filebeat/input/s3/input.go]: Is there any constant for this in `awssdk`?

File: x-pack/filebeat/input/s3/input.go [hunk 2]
```diff
@@ -178,30 +178,45 @@ func (p *s3Input) Run() {
...
+					p.logger.Warn(errors.Wrap(err, "credentials are invalid, acquiring AWS credentials again from config"))
+
+					awsConfig, err := awscommon.GetAWSCredentials(p.config.AwsConfig)
```
Comment (exekias) [inline x-pack/filebeat/input/s3/input.go]: is it possible to predict when the credentials will expire instead of waiting for the error?

Comment (kaiyan-sheng) [reply x-pack/filebeat/input/s3/input.go]: I haven't found a way to proactively check this. With the shared credential profile, we lose the expiration timestamp information. The expiration timestamp is generated when we call the `GetSessionTokenRequest` manually to generate a temporary credential. But this info is not stored in the shared credential profile. So after this `awscommon.GetAWSCredentials` function, the awsconfig shows `Expires =  0001-01-01 00:00:00 +0000 UTC` and `CanExpire =  false`.

File: x-pack/filebeat/input/s3/input.go (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/filebeat/input/s3/input.go]: There is not, but while looking for constant in `awssdk`, I found `IsErrorExpiredCreds` this function to check if the error is `ExpiredToken`. This does not cover `InvalidClientTokenId ` but I think we are ok here. `InvalidClientTokenId ` only happens when you manually changed `aws_access_key_id` in shared credential profile file.

**Comment by kaiyan-sheng** [comment]
@exekias This change won't affect metricbeat but metricbeat unfortunately has the same issue. I'm thinking to open a separate PR for it because the change will be in each metricset's code I think.

**Pull request merged**